public/js/dbkjs/layers.js

```diff
@@ -162,13 +162,52 @@ dbkjs.layers = {
             // TODO ArcGIS93Rest
         });
     },
-
+    switchedBaseLayer: null,
+    switchedZoomBaseLayer: null,
+    switchedAtZoomLevel: null,
     createBaseLayers: function() {
         var baselayer_ul = $('<ul id="baselayerpanel_ul" class="nav nav-pills nav-stacked">');
+
         $.each(dbkjs.options.baselayers, function(bl_index, bl) {
             var _li = $('<li class="bl" id="bl' + bl_index + '"><a href="#">' + bl.name + '</a></li>');
             baselayer_ul.append(_li);
-            bl.events.register("loadstart", bl, function() {
+            bl.events.register("loadstart", bl, function(e) {
+                // Change baselayer to options.afterMaxZoomLevelSwitchToLayer based on options.switchAtZoomLevel                
+                if (bl.options.afterMaxZoomLevelSwitchToLayer || dbkjs.layers.switchedZoomBaseLayer) {
+                    var zl = dbkjs.map.getZoom();
+                    if (zl === bl.options.switchAtZoomLevel ?? 10) {
+                        if (!dbkjs.layers.switchedZoomBaseLayer) {
+                            dbkjs.map.setBaseLayer(dbkjs.map.getLayersByName(bl.options.afterMaxZoomLevelSwitchToLayer)[0]);
+                            dbkjs.layers.switchedZoomBaseLayer = bl;
+                            dbkjs.layers.switchedAtZoomLevel = bl.options.switchAtZoomLevel ?? 10
+                        }
+                    } else {
+                        if (!bl.options.afterMaxZoomLevelSwitchToLayer && zl === dbkjs.layers.switchedAtZoomLevel - 1) {
+                            dbkjs.map.setBaseLayer(dbkjs.layers.switchedZoomBaseLayer);
+                            dbkjs.layers.switchedZoomBaseLayer = null;
+                            dbkjs.layers.switchedAtZoomLevel = null;
+                        }
+                    }
+                }
+                // Change baselayer to options.outsideOrganisationExtentSwitchToLayer based on 
+                // dbkjs.options.organisationExtent + dbkjs.options.organisationExtentBounderyInMeters
+                if (dbkjs.options.organisationExtent && (bl.options.outsideOrganisationExtentSwitchToLayer || dbkjs.layers.switchedBaseLayer)) {
+                    var extent = new OpenLayers.Format.GeoJSON().read(dbkjs.options.organisationExtent);
```

Comment:
TBH I don't like this solution, I'd prefer using a tileLoadFunction which fallbacks on load errors